amd/src/ui.js

```diff
@@ -14,60 +14,75 @@
 // along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
 
 /**
- * Tiny C4L UI.
+ * Tiny Elements UI.
  *
- * @module      tiny_c4l/ui
+ * @module      tiny_elements/ui
  * @copyright   2022 Marc Català <reskit@gmail.com>
  * @license     http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
  */
 
-import {component} from './common';
-import C4LModal from './modal';
+import ElementsModal from './modal';
 import ModalFactory from 'core/modal_factory';
-import {components as Components} from './components';
-import {get_strings as getStrings} from 'core/str';
 import {
     isStudent,
-    getallowedComponents,
     showPreview,
-    getpreviewCSS,
-    getcustomComponents
+    canManage,
+    markComponents
 } from './options';
 import ModalEvents from 'core/modal_events';
 import {
     addVariant,
     getVariantsClass,
     getVariantHtml,
+    getVariantPreferences,
     getVariantsHtml,
     loadVariantPreferences,
     removeVariant,
-    saveVariantPreferences,
-    variantExists
+    setData as setVariantsData
 } from './variantslib';
-
-let userStudent = false;
-let previewC4L = true;
-let allowedComponents = [];
-let Contexts = [];
-let langStrings = {};
-let previewCSS = '';
-let customComponents = [];
-const compPrefix = 'c4lv-';
+import {
+    savePreferences,
+    loadPreferences,
+    Preferences
+} from './preferencelib';
+import {getContextId} from 'editor_tiny/options';
+import Data from './data';
+
+let currentFlavor = '';
+let currentFlavorId = 0;
+let currentCategoryId = 1;
+let currentCategoryName = '';
+let lastFlavor = [];
+let sel = '';
+let data = {};
 
 /**
  * Handle action
  *
  * @param {TinyMCE} editor
  */
 export const handleAction = async(editor) => {
-    userStudent = isStudent(editor);
-    previewC4L = showPreview(editor);
-    customComponents = getcustomComponents(editor);
-    addCustomComponents();
-    allowedComponents = getallowedComponents(editor);
-    previewCSS = getpreviewCSS(editor);
-    langStrings = await getAllStrings();
-    loadVariantPreferences(Components).then(() => displayDialogue(editor));
+    sel = editor.selection.getContent();
```

Comment:
Telling variable names pls :)

```suggestion
    selection = editor.selection.getContent();
```